docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1619,359 +1635,597 @@ We **strongly recommend** using these markers any time when you use an extended
-div:has-text(/test-xpath-content/):xpath(../../..)
-// Use xpath only to select elements
-facebook.com##:xpath(//div[@id="stream_pagelet"]//div[starts-with(@id,"hyperfeed_story_id_")][.//h6//span/text()="People You May Know"])
+
+`:xpath(//*[@class="inner"]/..)` selects the element `div#target2`:
+```html
+<!-- HTML code -->
+<div id="target2">
+  <div class="inner"></div>
+</div>
 ```
 
-##### Pseudo-class `:nth-ancestor()`
 
-This pseudo-class allows to lookup the nth ancestor relative to the currently selected node.
+#### Pseudo-class `:nth-ancestor()` {#extended-css-nth-ancestor}
 
-It is a low-overhead equivalent to `:xpath(..[/..]*)`.
+The `:nth-ancestor()` pseudo-class allows to lookup the *nth* ancestor relative to the previously selected element.
 
-> **Can be placed only at the end of a selector, except for [pseudo-class `:remove()`](#remove-pseudos).**
+```
+subject:nth-ancestor(n)
+```
+- `subject` — required, standard or extended css selector
+- `n` — required, number >= 1 and < 256, distance to the needed ancestor from the element selected by `subject`
 
 **Syntax**
+
 ```
-selector:nth-ancestor(n)
+subject:nth-ancestor(n)
 ```
-- `selector` — a plain CSS selector, or a Sizzle compatible selector.
-- `n` — positive number >= 1 and < 256, distance from the currently selected node.
+- `subject` — required, standard or extended css selector
+- `n` — required, number >= 1 and < 256, distance to the needed ancestor from the element selected by `subject`
+
+##### `:nth-ancestor()` limitations {#extended-css-nth-ancestor-limitations}
+
+> The `:nth-ancestor()` pseudo-class is not supported inside argument of the [`:not()` pseudo-class](#extended-css-not).
 
 **Examples**
-```
-div.test:nth-ancestor(4)
-div:has-text(/test/):nth-ancestor(2)
+
+For such DOM:
+```html
+<!-- HTML code -->
+<div id="target1">
+  <div class="child"></div>
+
+  <div id="target2">
+    <div>
+      <div>
+        <div class="inner"></div>
+      </div>
+    </div>
+  </div>
+</div>
 ```
 
-##### Pseudo-class `:upward()`
+`.child:nth-ancestor(1)` selects the element `div#target1`,
+`div[class="inner"]:nth-ancestor(3)` selects the element `div#target2`.
 
-This pseudo-class allows to lookup the ancestor relative to the currently selected node.
 
-> **Can be placed only at the end of a selector, except for [pseudo-class `:remove()`](#remove-pseudos).**
+#### Pseudo-class `:upward()` {#extended-css-upward}
+
+The `:upward()` pseudo-class allows to lookup the ancestor relative to the previously selected element.
 
 **Syntax**
+
 ```
-/* selector parameter */
-subjectSelector:upward(targetSelector)
-/* number parameter */
-subjectSelector:upward(n)
+subject:upward(ancestor)
 ```
-- `subjectSelector` — a plain CSS selector, or a Sizzle compatible selector
-- `targetSelector` — a valid plain CSS selector
-- `n` — positive number >= 1 and < 256, distance from the currently selected node
+- `subject` — required, standard or extended css selector
```

Comment:
CSS capitalized, same for all other instances when CSS is used outside of ExtendedCss or other similar cases